### PR TITLE
Deprecate ajax call to search action

### LINKED ISSUE #7259: Deprecate `AdminInterface::setRequest` and inject RequestStack instead.
"sonata-project/admin-bundle": "^3.98"

Search inside admin entities didn't initialize request before asking configureDefaultFilterValues, so if I had the multi language admin, I can't preset the locale filter:

```
protected function configureDefaultFilterValues(array &$filterValues) 
    {
        $filterValues['locale'] = [
            'value' => $this->request->getLocale(),
        ];
    }
```

it makes - Call to a member function getLocale() on null - on admin/search?q=test request.

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

BC

Closes: #7168

## Changelog

```markdown
### Deprecated
- Passing `admin` in the request of search action.
```

### CODE REVIEW

File: src/Action/SearchAction.php [hunk 1]
```diff
@@ -78,7 +96,18 @@ public function __construct(
      */
     public function __invoke(Request $request): Response
     {
-        if (!$request->get('admin') || !$request->isXmlHttpRequest()) {
+        // NEXT_MAJOR: Remove this BC-layer.
+        if (null === $request->get('_sonata_admin') && null !== $request->get('admin')) {
```
Comment (VincentLanglet) [inline src/Action/SearchAction.php:100]: I never find in the code where the `admin` value is passed to the request.

Any idea @sonata-project/contributors ? I need to pass `_sonata_admin` instead to fix the deprecation on 3.x.

**Comment by VincentLanglet** [comment]
I thought this was closing #7259, but seems like the admin block are rendered by 
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Block/AdminSearchBlockService.php#L143-L187 instead.

**Comment by VincentLanglet** [comment]
The search action is saying

> The search action first render an empty page, if the query is set, then the template generates
     some ajax request to retrieve results for each admin. The Ajax query returns a JSON response.

But I didn't find where the ajax call is made...

File: src/Action/SearchAction.php (hunk 1 above)
Comment (franmomu) [reply src/Action/SearchAction.php:100]: hmm looks like it is supposed to be here:

https://github.com/sonata-project/SonataAdminBundle/blob/5e9b18a0fd548b1b26d312da0c2d70d0df0e7dcd/src/Resources/views/standard_layout.html.twig#L201-L210

but it doesn't and apparently it wasn't added at the beginning: https://github.com/sonata-project/SonataAdminBundle/commit/53dd431540fd9173bda6cb03d53c80d4fb030cf7

So I would say it is something we've never used.

Comment (VincentLanglet) [reply src/Action/SearchAction.php:100]: I will try to deprecate it instead then.

**Comment by VincentLanglet** [comment]
@sonata-project/contributors Since the ability to pass an admin to the search action for an ajax call is never used, I think we should deprecate it in order to avoid duplicating the logic of the AdminSearchBlockService.

**Review by jordisala1991 — APPROVED** [review]

**Review by phansys — APPROVED** [review]

**Pull request merged**